kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/internal/parquet/ParquetSchemaUtils.java

```diff
@@ -93,33 +124,51 @@ static Type findSubFieldType(
     static Map<Integer, Type> getParquetFieldToTypeMap(GroupType parquetGroupType) {
         // Generate the field id to Parquet type map only if the read schema has field ids.
         return parquetGroupType.getFields().stream()
-            .filter(subFieldType -> subFieldType.getId() != null)
-            .collect(
-                Collectors.toMap(
-                    subFieldType -> subFieldType.getId().intValue(),
-                    subFieldType -> subFieldType,
-                    (u, v) -> {
-                        throw new IllegalStateException(String.format("Parquet file contains " +
-                            "multiple columns (%s, %s) with the same field id", u, v));
-                    }));
+                .filter(subFieldType -> subFieldType.getId() != null)
+                .collect(Collectors.toMap(
+                        subFieldType -> subFieldType.getId().intValue(),
+                        subFieldType -> subFieldType,
+                        (u, v) -> {
+                            throw new IllegalStateException(
+                                    format("Parquet file contains multiple columns " +
+                                            "(%s, %s) with the same field id", u, v));
+                        }));
+    }
+
+    /**
+     * Convert the given Kernel schema to Parquet's schema
+     *
+     * @param structType Kernel schema object
+     * @return {@link MessageType} representing the schema in Parquet format.
+     */
+    public static MessageType toParquetSchema(StructType structType) {
+        List<Type> types = new ArrayList<>();
+        for (StructField structField : structType.fields()) {
+            types.add(toParquetType(
+                    structField.getDataType(),
+                    structField.getName(),
+                    structField.isNullable() ? OPTIONAL : REQUIRED,
+                    getFieldId(structField)));
+        }
+        return new MessageType("Default Kernel Schema", types);
     }
 
     private static List<Type> pruneFields(
-        GroupType type, StructType deltaDataType, boolean hasFieldIds) {
+            GroupType type, StructType deltaDataType, boolean hasFieldIds) {
         // prune fields including nested pruning like in pruneSchema
         final Map<Integer, Type> parquetFieldIdToTypeMap = getParquetFieldToTypeMap(type);
 
         return deltaDataType.fields().stream()
-            .map(column -> {
-                Type subType = findSubFieldType(type, column, parquetFieldIdToTypeMap);
-                if (subType != null) {
-                    return prunedType(subType, column.getDataType(), hasFieldIds);
-                } else {
-                    return null;
-                }
-            })
-            .filter(Objects::nonNull)
-            .collect(Collectors.toList());
+                .map(column -> {
```

Comment:
is this just indentation change? can we ignore this noise?

Reply:
It got changed as part of the autoformat in IntelliJ. Just space changes are easy to visualize in github.